cms/djangoapps/contentstore/tests/test_libraries.py

```diff
@@ -306,3 +325,268 @@ def test_change_after_first_sync(self):
+                'parent_locator': unicode(self.library.location), 'category': 'html',
+            })
+            self.assertIn(response.status_code, (200, 403))  # 400 would be ambiguous
+            return response.status_code == 200
+
+        # Check that we do not have read or write access to block:
+        self.assertFalse(can_read_block())
+        self.assertFalse(can_edit_block())
+        self.assertFalse(can_delete_block())
+        self.assertFalse(can_copy_block())
+        self.assertFalse(can_create_block())
+
+        # Give non_staff_user read-only permission:
+        if use_org_level_role:
+            OrgLibraryUserRole(self.lib_key.org).add_users(self.non_staff_user)
+        else:
+            LibraryUserRole(self.lib_key).add_users(self.non_staff_user)
+
+        self.assertTrue(self._can_access_library(self.library))
+        self.assertTrue(can_read_block())
+        self.assertFalse(can_edit_block())
+        self.assertFalse(can_delete_block())
+        self.assertFalse(can_copy_block())
+        self.assertFalse(can_create_block())
+
+    @ddt.data(
+        (LibraryUserRole, CourseStaffRole, True),
+        (CourseStaffRole, CourseStaffRole, True),
+        (None, CourseStaffRole, False),
+        (LibraryUserRole, None, False),
+    )
+    @ddt.unpack
+    def test_duplicate_across_courses(self, library_role, course_role, expected_result):
+        """
+        Test that the REST API will correctly allow/refuse when copying
+        from a library with (write, read, or no) access to a course with (write or no) access.
+        """
+        # As staff user, add a block to self.library:
+        block = ItemFactory.create(category="html", parent_location=self.library.location, user_id=self.user.id, publish_item=False)
+        # And create a course:
+        with modulestore().default_store(ModuleStoreEnum.Type.split):
+            course = CourseFactory.create()
+
+        self._login_as_non_staff_user()
+
+        # Assign roles:
+        if library_role:
+            library_role(self.lib_key).add_users(self.non_staff_user)
+        if course_role:
+            course_role(course.location.course_key).add_users(self.non_staff_user)
+
+        # Copy block to the course:
+        response = self.client.ajax_post(reverse_url('xblock_handler'), {
+            'parent_locator': unicode(course.location),
+            'duplicate_source_locator': unicode(block.location),
+        })
+        self.assertIn(response.status_code, (200, 403))  # 400 would be ambiguous
+        duplicate_action_allowed = (response.status_code == 200)
+        self.assertEqual(duplicate_action_allowed, expected_result)
+
+    @ddt.data(
+        (LibraryUserRole, CourseStaffRole, True),
+        (CourseStaffRole, CourseStaffRole, True),
+        (None, CourseStaffRole, False),
+        (LibraryUserRole, None, False),
+    )
+    @ddt.unpack
+    def test_refresh_library_content_permissions(self, library_role, course_role, expected_result):
```

Comment:
Yes, it will stop you from saving the changes to the XBlock if you try to enter in a library for which you do not have permission. It also will not allow you to refresh the XBlock if someone else has configured it to reference a library that you do not have access to.